source/gfoidl.Stochastics/Statistics/Sample.Delta.cs

```diff
@@ -58,42 +30,48 @@ internal double CalculateDeltaParallelizedSimd()
                 Partitioner.Create(0, _values.Length),
                 range =>
                 {
-                    double localDelta = 0;
-                    double[] arr      = _values;
-                    double avg        = this.Mean;
-                    int i             = range.Item1;
+                    double localDelta = this.CalculateDeltaImpl((range.Item1, range.Item2));
 
-                    // RCE
-                    if ((uint)range.Item1 >= arr.Length || (uint)range.Item2 > arr.Length)
-                        ThrowHelper.ThrowArgumentOutOfRange(nameof(range));
+                    lock (sync) delta += localDelta;
```

Comment:
Maybe `Interlocked.Add`?

Reply:
Argh, Interlocked.Add has only overloads for `int` and `long`. But this can be written with `Interlocked.CompareExchange`.